perceval/serialization/deserialize.py

```diff
@@ -407,3 +424,25 @@ def resolve(self):
             experiment.set_postselection(deserialize(post_select_str))
 
         return experiment
+
+
+class CompiledCircuitBuilder:
```

Comment:
I'm not sure that you particularly need this class. A regular function would be enough (the other builders need dictionaries containing methods, that's why they are here).

Also, moving the logic of this method elsewhere could be good (I think that `component_deserialization.py` could be a good spot)